homeassistant/components/knx/climate.py

```diff
@@ -296,21 +299,33 @@ def hvac_modes(self) -> Optional[List[str]]:
         ]
 
         if self.device.supports_on_off:
-            _operations.append(HVAC_MODE_HEAT)
             _operations.append(HVAC_MODE_OFF)
 
         _modes = list(filter(None, _operations))
         # default to ["heat"]
         return _modes if _modes else [HVAC_MODE_HEAT]
-
+    
+    @property
+    def hvac_action(self):
+        """The current HVAC action (heating, cooling)"""
+        mode = self.hvac_mode
+        if mode == HVAC_MODE_FAN_ONLY:
+            return CURRENT_HVAC_IDLE
+        if mode == HVAC_MODE_HEAT:
+            return CURRENT_HVAC_HEAT
+        if mode == HVAC_MODE_COOL:
+            return CURRENT_HVAC_COOL
+        if mode == HVAC_MODE_DRY:
+            return CURRENT_HVAC_DRY
+        return CURRENT_HVAC_OFF
+    
     async def async_set_hvac_mode(self, hvac_mode: str) -> None:
         """Set operation mode."""
         if self.device.supports_on_off and hvac_mode == HVAC_MODE_OFF:
             await self.device.turn_off()
-        elif self.device.supports_on_off and hvac_mode == HVAC_MODE_HEAT:
-            await self.device.turn_on()
```

Comment:
devices that support only on/off but don't support any operation_mode couldn't be turned on then.